tests/test_perf.py

```diff
@@ -144,6 +145,7 @@ def test_perf_comm_change(
     I'm not sure it can be done, i.e is this info even kept anywhere).
     """
     with system_profiler as profiler:
+        time.sleep(2)
```

Comment:
Why are the sleeps necessary? Is there a way to make the test deterministic instead?